locations/commands/nsi.py

```diff
@@ -36,6 +37,12 @@ def add_options(self, parser):
             action="store_true",
             help="Query NSI for wikidata code",
         )
+        parser.add_argument(
+            "--detect-missing",
+            dest="detect_missing",
+            action="store_true",
+            help="Query NSI for missing by NSI category. ie brands/shop/supermarket",
```

Comment:
I thought about that, and the existing code does fetch a compressed wikidata listing.

I went for this approach so that you could opt in that what you were checking - I don't know that all of the NSI warrants trying to build a spider. It may do, but my particular focus was on /brand/shop/* - and potentially it was a human scale problem to check all missing /brand/shop/art (6 entries) as a opposed to everything.

The good news is if you want to do everything in NSI, the above approach would need minimal refactoring to support it.

Reply:
(to be clear, my main intent here is to raise useful GitHub issues for development. Coupled with the "detect store finder" ideas, it has a chance to generate a lot of high quality linked data with lower effort than bespoke spiders and research)

Reply:
Hmm, maybe have some easy way to list what is in NSI? I mostly though about how to make easier to find correct codes without manually inspecting NSI. And be able to somehow notice NSI adding entry with say a new shop value

Reply:
> be able to somehow notice NSI adding entry with say a new shop value

So ideally, this will end up giving us that.

My (very manual) process at the moment is:

- Run for a category, create a discusson
- Work through and find things with storefinders
- Raise github issue
- Ideally, that will get turned into a scraper at some point

That will leave then this tool picking up

- Things without scrapers
- Things that cannot have scrapers

For the things that cannot have scrapers, there's the potential for adding some kind of ignore list, or link to an issue/discussion where it is discussed as to why it can't be done.

If we get to that point, then this would achieve the goal of "Know when a new brand is in the NSI".

Reply:
Well, two weeks later:

- All NSI brand/shop/* have been covered, except for window blinds and similar I didn't want to cover
- There's a bit more to do with auto detection of spiders before the next run happens
- I think everyone is sick of seeing feature requests for Wordpress spiders!